File: fixtures/typescript/negative_different_control_flow/right.ts
export function classifyInvoice(invoice: Invoice): State {
  switch (invoice.status) {
    case Status.Void:
      return State.Cancelled;
    case Status.Settled:
      return invoice.amount > 0 ? State.Closed : State.Open;
    default:
      return State.Open;
  }
}

File: fixtures/rust/negative_different_control_flow/right.rs
pub fn classify_receipt(receipt: &Receipt) -> State {
    match receipt.status {
        Status::Void => State::Cancelled,
        Status::Settled if receipt.amount > 0 => State::Closed,
        Status::Settled => State::Open,
        Status::Pending => State::Open,
    }
}
